### PR TITLE
server: use UTC timestamps in dex.

### PR DESCRIPTION
**Pull request by rithujohn191**
Updated server now() func and logger to use UTC timestamps.

Log messages now use UTC:
time="2016-12-16T22:44:47Z" level=info msg="config issuer: http://127.0.0.1:5556/dex" 
time="2016-12-16T22:44:47Z" level=info msg="config connector: mock" 
time="2016-12-16T22:44:47Z" level=info msg="config connector: local passwords enabled" 
time="2016-12-16T22:44:47Z" level=info msg="config storage: sqlite3" 
time="2016-12-16T22:44:47Z" level=info msg="config static client: example-app" 
time="2016-12-16T22:44:47Z" level=info msg="listening (http) on 0.0.0.0:5556" 

@ericchiang

### CODE REVIEW

File: cmd/dex/serve.go [hunk 1]
```diff
@@ -251,6 +255,15 @@ var (
 	logFormats = []string{"json", "text"}
 )
 
+type customFormatter struct {
```
Comment (ericchiang) [inline cmd/dex/serve.go]: utcFormatter?

File: server/server_test.go [hunk 2]
```diff
@@ -159,7 +159,7 @@ func TestOAuth2CodeFlow(t *testing.T) {
 	clientSecret := "testclientsecret"
 	requestedScopes := []string{oidc.ScopeOpenID, "email", "profile", "groups", "offline_access"}
 
-	t0 := time.Now()
+	t0 := time.Now().UTC()
```
Comment (ericchiang) [inline server/server_test.go]: Maybe we should change our comparison rather than our initialization?

Comment (rithujohn191) [inline server/server_test.go]: This value gets passed on to the now() func which gets used in newKeyCacher(). So this flow would mimic the actual behavior of the server now() function.

File: cmd/dex/serve.go (hunk 1 above)
Comment (rithujohn191) [inline cmd/dex/serve.go]: done

File: server/server_test.go (hunk 2 above)
Comment (rithujohn191) [inline server/server_test.go]: Discussed offline. Removing the UTC to increase test coverage to include cases when time zones are involved.

**Comment by ericchiang** [comment]
lgtm

**Pull request merged**